backend/controller/stock.py

```diff
@@ -20,4 +20,38 @@ def getUserSavedStocks(self, userid):
         result = []
         for stock in stocks:
             result.append(self.build_saved_stocks_dict(stock))
-        return jsonify(result), 200
\ No newline at end of file
+        return jsonify(result), 200
+
+
+    def saveUserStock(self, json, userid):
+        symbol = json['symbol']
+        dao = StockDAO()
+        stock = dao.getStockBySymbol(symbol)
+
+        if stock:
+            stockid = stock[0]
+        else:
+            stockid = dao.createStock(symbol)
+
+        saved = dao.saveUserStock(userid, stockid)
+        dao.conn.close()
+
+        if saved:
+            return jsonify("STOCK WAS SAVED SUCCESSFULLY."), 201
+        else:
+            return jsonify("EITHER STOCK WAS ALREADY SAVED OR USER DOESN'T EXIST."), 500
```

Comment:
I think we can be a little more generic on the message. We are giving too much information by saying the cause of the error that is not really necesary. Something like "STOCK WAS NOT SAVED SUCCESFULLY" or in that matter is enough. Also the status code maybe 400 is better since 500 is an internal error meaning something unexpected happened which is not exactly the case?

Reply:
Done.